### PR TITLE
tls: add test for blocked TLS handshake

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Add a test to ensure that we can't wedge the cockpit-tls server by
blocking during the handshake phase (which could even theoretically
happen by accident if a connection is dropped, for example).

This was a bug before moving to the threaded connection model.  The
lock-up eventually cleared when hitting the gnutls handshake timeout,
but 40 seconds is a long time for the server to be blocked from handling
any other incoming requests.

### CODE REVIEW

File: src/tls/test-server.c [hunk 1]
```diff
@@ -345,6 +346,49 @@ test_no_tls_many_serial (TestCase *tc, gconstpointer data)
...
+      /* make sure we can do a second connection while the first one is
+       * blocked in the handshake
+       */
+      gint second_fd = do_connect (tc);
+      send_request (second_fd, "GET / HTTP/1.1\r\nHost: localhost\r\n\r\n");
+
+      /* wait 10 seconds for the reply */
+      struct pollfd pfd = { .fd = second_fd, .events = POLLIN };
+      g_assert_cmpint (poll (&pfd, 1, 10000), ==, 1);
+      close (second_fd);
```
Comment (martinpitt) [inline src/tls/test-server.c]: Couldn't this just use `assert_http()` here?

Comment (allisonkarlitskaya) [reply src/tls/test-server.c]: No.  This is what I tried first, but this calls `do_request()` which iterates the server's mainloop from the calling thread (thus unblocking things).

**Comment by martinpitt** [comment]
I put this into https://github.com/allisonkarlitskaya/cockpit/commits/tls-priv which we'll land as one PR. So we don't need this PR any more.

**Comment by allisonkarlitskaya** [comment]
I took it out of the branch because it's unrelated.

**Comment by martinpitt** [comment]
I undusted this and rebased to current master. I changed the commit message to adjust to the now fixed situation.

**Pull request merged**